core/dbt/exceptions.py

```diff
@@ -2145,14 +2147,65 @@ def __init__(self, yaml_columns, sql_columns):
         self.sql_columns = sql_columns
         super().__init__(msg=self.get_message())
 
-    def get_message(self) -> str:
+    def get_mismatches(self) -> agate.Table:
+        # avoid a circular import
+        from dbt.clients.agate_helper import table_from_data_flat
+
+        column_names = ["column_name", "definition_type", "contract_type", "mismatch_reason"]
+        # list of mismatches
+        mismatches: List[Dict[str, str]] = []
+        # track sql cols so we don't need another for loop later
+        sql_col_set = set()
+        # for each sql col list
+        for sql_col in self.sql_columns:
+            # add sql col to set
+            sql_col_set.add(sql_col["name"])
+            # for each yaml col list
+            for i, yaml_col in enumerate(self.yaml_columns):
+                # if name matches
+                if sql_col["name"] == yaml_col["name"]:
+                    # if type matches
+                    if sql_col["formatted"] == yaml_col["formatted"]:
```

Comment:
for my own understanding - any reason not to do the comparison on data_type at this point?